include/mbedtls/ssl.h

```diff
@@ -802,6 +802,15 @@ typedef struct mbedtls_ssl_flight_item mbedtls_ssl_flight_item;
 #endif
 
 #if defined(MBEDTLS_SSL_EARLY_DATA) && defined(MBEDTLS_SSL_CLI_C)
+/* Define the status of early data.
```

Comment:
I plan to push the PR addressing #4902 this week.

> The statuses looks not good enough for me

Could you elaborate ?

Reply:
Just a feeling, I am not sure if my idea is correctly.  I will post my points to #4902.

Reply:
My idea is about internal states on server side. It looks like #4902 is not right place.

- NOT_RECEIVED: Initial state
- REJECTED:   Reject the following up early app data. In this state, decrypt error will be ignore. 
- ACCEPTED:  Accept the following up early app data. In this state, early_data indication will sent. 

State machine
- NOT_RECEIVED -> REJECTED : `early_data extension` is received (ClientHello)
- REJECTED->ACCEPTED: If early data is allowed(valid ticket, early_data_enabled and so on), otherwise do not change it. this step should be done in `post_client_hello`.

For client side, I guess 3 states are enough.

- NOT_SENT: Initial state
- REJECTED: If early data indication is sent, change to it. Early app data can be sent in this state before `CLIENT_FINISHED`. [RFC8446 A.2](https://datatracker.ietf.org/doc/html/rfc8446#appendix-A)
- ACCEPTED: If early data is received in EE, change to it.

Reply:
In the prototype `early_data_status` is used and defined only on the client side thus mechanically I have not thought about it in the server context. I am not saying that we shouldn't, this seems a sensible idea. This PR is client specific thus I think we can postpone this discussion though.

On client side as I would like to merge in one state/status the `early_data` and `early_data_status` state/status of the prototype, three states are not enough I think. I will describe them in the PR addressing #4902.

Reply:
Sure, we should postpone the discussion.